confgenerator/config.go

```diff
@@ -82,109 +72,452 @@ func ParseUnifiedConfigAndValidate(input []byte, platform string) (UnifiedConfig
 	return config, nil
 }
 
+type component interface {
+	Type() string
+	ValidateType(subagent string, component string, id string, platform string) error
+	ValidateParameters(subagent string, component string, id string) error
```

Comment:
Validation shouldn't be a required part of the interface; it should be optional, only if the validation cannot be accomplished with struct tags alone.

Reply:
This is in part due to the validation being specific to the platform that the config is generated for. Using a library that would allow passing a context object through would remove the need for separate functions in the interface.

Reply:
The validation library I linked above allows that.

Reply:
But the yaml library doesn't seem to — at least I didn't find a way to propagate the context into the unmarshaler. Looking forward to seeing your prototype.